lib/src/rules/diagnostic_describe_all_properties.dart

```diff
@@ -0,0 +1,248 @@
+    return inheritanceManager.getInherited(
+            classElement.type, new Name(libraryUri, member.name)) !=
+        null;
+  }
+
+  @override
+  visitClassDeclaration(ClassDeclaration node) {
+    ++totalClasses;
+
+    // We only care about Diagnosticables.
+    var type = node.declaredElement.type;
+    if (!DartTypeUtilities.implementsInterface(type, 'Diagnosticable', '')) {
+      return;
+    }
+
+    var properties = <SimpleIdentifier>[];
+    for (var member in node.members) {
+      if (member is MethodDeclaration && member.isGetter) {
+        if (isPrivate(member.name)) {
+          continue;
+        }
+        if (member.isStatic) {
+          continue;
+        }
+        if (isWidgetProperty(member.returnType?.type)) {
+          continue;
+        }
+        if (_isOverridingMember(member.declaredElement)) {
+          continue;
+        }
+
+        properties.add(member.name);
+      } else if (member is FieldDeclaration) {
+        for (var v in member.fields.variables) {
+          if (isPrivate(v.name)) {
+            continue;
+          }
+          if (v.declaredElement.isStatic) {
+            continue;
+          }
+          if (isWidgetProperty(v.declaredElement.type)) {
+            continue;
+          }
+          if (_isOverridingMember(v.declaredElement)) {
+            continue;
+          }
+
+          properties.add(v.name);
+        }
+      }
+    }
+
+    if (properties.isEmpty) {
+      return;
+    }
+
+    // todo (pq): move up to top when we're not counting anymore.
+    var debugFillProperties = node.getMethod('debugFillProperties');
+    if (debugFillProperties == null) {
+      ++classesWithPropertiesButNoDebugFill;
+    }
+
+    var debugDescribeChildren = node.getMethod('debugDescribeChildren');
+    if (debugFillProperties == null && debugDescribeChildren == null) {
+      return;
+    }
+
+    // Remove any defined in debugFillProperties.
+    removeReferences(debugFillProperties, properties);
+
+    // Remove any defined in debugDescribeChildren.
+    removeReferences(debugDescribeChildren, properties);
+
+    // Flag the rest.
+    properties.forEach(rule.reportLint);
+
+//    // todo (pq): remove before landing
+//    for (var prop in properties) {
+//      var line = lineInfo.getLocation(prop.offset).lineNumber;
+//      var prefix =
+//          'https://github.com/flutter/flutter/blob/master/packages/flutter/';
+//      var path = node.element.source.fullName.split('packages/flutter/')[1];
+//      print('| [$path:$line]($prefix$path#L$line) | ${node.name}.$prop |');
+//      ++debugPropertyCount;
+//    }
+  }
+
+  void removeReferences(
+      MethodDeclaration method, List<SimpleIdentifier> properties) {
+    if (method == null) {
+      return;
+    }
+    DartTypeUtilities.traverseNodesInDFS(method.body)
+        .whereType<SimpleIdentifier>()
+        .forEach((p) {
+      var debugName;
+      var name;
+      if (p.name.startsWith('debug') && p.name.length > 5) {
```

Comment:
nit: make a string const
const debugPrefix = 'debug'
and then use debugPrefix.length in the few cases it is used.